internal/db/schema/postgres/postgres_test.go

```diff
@@ -33,490 +33,352 @@ package postgres
 import (
 	"bytes"
 	"context"
-	"database/sql"
-	sqldriver "database/sql/driver"
 	"fmt"
-	"io"
-	"log"
 	"strconv"
 	"strings"
 	"testing"
 
-	"github.com/dhui/dktest"
-	"github.com/golang-migrate/migrate/v4/dktesting"
+	"github.com/hashicorp/boundary/testing/dbtest"
 	"github.com/stretchr/testify/assert"
 	"github.com/stretchr/testify/require"
 )
 
-const (
-	pgPassword = "postgres"
-)
-
-var (
-	opts = dktest.Options{
-		Env:          map[string]string{"POSTGRES_PASSWORD": pgPassword},
-		PortRequired: true, ReadyFunc: isReady,
-	}
-	// Supported versions: https://www.postgresql.org/support/versioning/
-	specs = []dktesting.ContainerSpec{
-		{ImageName: "postgres:12", Options: opts},
-	}
-)
-
-func pgConnectionString(host, port string) string {
-	return fmt.Sprintf("postgres://postgres:%s@%s:%s/postgres?sslmode=disable", pgPassword, host, port)
-}
-
-func isReady(ctx context.Context, c dktest.ContainerInfo) bool {
-	ip, port, err := c.FirstPort()
-	if err != nil {
-		return false
-	}
-
-	db, err := sql.Open("postgres", pgConnectionString(ip, port))
-	if err != nil {
-		return false
-	}
-	defer func() {
-		if err := db.Close(); err != nil {
-			log.Println("close error:", err)
-		}
-	}()
-	if err = db.PingContext(ctx); err != nil {
-		switch err {
-		case sqldriver.ErrBadConn, io.EOF:
-			return false
-		default:
-			log.Println(err)
-		}
-		return false
-	}
-
-	return true
-}
-
 func TestDbStuff(t *testing.T) {
-	dktesting.ParallelTest(t, specs, func(t *testing.T, c dktest.ContainerInfo) {
-		ctx := context.Background()
-		ip, port, err := c.FirstPort()
-		if err != nil {
-			t.Fatal(err)
-		}
-
-		addr := pgConnectionString(ip, port)
-		d, err := open(t, ctx, addr)
-		if err != nil {
-			t.Fatal(err)
-		}
-		defer func() {
-			if err := d.close(t); err != nil {
-				t.Error(err)
-			}
-		}()
-		test(t, d, []byte("SELECT 1"))
+	t.Parallel()
+	ctx := context.Background()
+	c, u, _, err := dbtest.StartUsingTemplate("postgres", dbtest.WithTemplate("template1"))
```

Comment:
Would you mind defining consts for the supported dialects and templates and using them in this PR?

Reply:
added consts and updated the tests to use them